interfaces/policy/policy_test.go

```diff
@@ -2316,3 +2361,116 @@ func (s *policySuite) TestSlotsArityAutoConnection(c *C) {
 		c.Check(arity.SlotsPerPlugAny(), Equals, t.any)
 	}
 }
+
+func (s *policySuite) TestNameConstraintsInstallation(c *C) {
+	const plugSnap = `name: install-snap
+version: 0
+plugs:
+  install-plug-name-bound:`
+
+	const plugOtherNameSnap = `name: install-snap
+version: 0
+plugs:
+  install-plug-name-bound-other:
+    interface: install-plug-name-bound
+`
+
+	const slotSnap = `name: install-snap
+version: 0
+slots:
+  install-slot-name-bound:`
+
+	const slotOtherNameSnap = `name: install-snap
+version: 0
+slots:
+  install-slot-name-bound-other:
+    interface: install-slot-name-bound
+`
+
+	const plugOtherName = `plugs:
+  install-plug-name-bound:
+    allow-installation:
+      plug-names:
+        - install-plug-name-bound-other`
+
+	tests := []struct {
+		installYaml string
+		plugsSlots  string
+		err         string // "" => no error
+	}{
+		{plugSnap, "", ""},
+		{plugOtherNameSnap, "", `installation not allowed by "install-plug-name-bound-other" plug rule of interface "install-plug-name-bound"`},
+		{plugOtherNameSnap, plugOtherName, ""},
+		{slotSnap, "", ""},
+		{slotOtherNameSnap, "", `installation not allowed by "install-slot-name-bound-other" slot rule of interface "install-slot-name-bound"`},
+	}
+
+	for _, t := range tests {
+		installSnap := snaptest.MockInfo(c, t.installYaml, nil)
+
+		plugsSlots := strings.TrimSpace(t.plugsSlots)
+		if plugsSlots != "" {
+			plugsSlots = "\n" + plugsSlots
+		}
+
+		a, err := asserts.Decode([]byte(strings.Replace(`type: snap-declaration
+authority-id: canonical
+series: 16
+snap-name: install-snap
+snap-id: installsnap6idididididididididid
+publisher-id: publisher
+@plugsSlots@
+timestamp: 2016-09-30T12:00:00Z
+sign-key-sha3-384: Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij
+
+AXNpZw==`, "\n@plugsSlots@", plugsSlots, 1)))
+		c.Assert(err, IsNil)
+		snapDecl := a.(*asserts.SnapDeclaration)
+
+		cand := policy.InstallCandidate{
+			Snap:            installSnap,
+			SnapDeclaration: snapDecl,
+			BaseDeclaration: s.baseDecl,
+		}
+		err = cand.Check()
+		if t.err == "" {
+			c.Check(err, IsNil)
+		} else {
+			c.Check(err, ErrorMatches, t.err)
+		}
+	}
+}
+
+func (s *policySuite) TestNameConstraintsAutoConnection(c *C) {
+	tests := []struct {
+		plug, slot string
+		ok         bool
+	}{
+		{"plugs-name-bound-p1", "plugs-name-bound-s1", false},
```

Comment:
I think this test would be more readable if instead of `ok book` we'd have `errorPattern string` and check for it using ErrorMatches. It would self-document this way. Alternatively add a comment to point the reader to https://github.com/snapcore/snapd/pull/7813/commits/d95b88908c2da52aae87b2510e4085b6f86cc39a#diff-6d929053445b2846e53d98c79665d924R525 as a hint as to where the yes-or-no rules come from.

Reply:
I did something in between